pkg/controller/certificatesigningrequests/util/signername_test.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestIssuerRefFromSignerName(t *testing.T) {
+	tests := map[string]struct {
+		inpName            string
+		expSignerIssuerRef SignerIssuerRef
+		expOK              bool
+	}{
+		"an empty name should return false": {
```

Comment:
suggestion: does this need test cases for issuers + clusterissuers with a "." in their names?

Reply:
Not sure I follow

Reply:
https://github.com/jetstack/cert-manager/pull/4064#discussion_r640683648 <- test cases for this.

Reply:
I see you added test cases for this so maybe you understood subconsciously :grin: